core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitUtil.java

```diff
@@ -0,0 +1,208 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.iceberg.actions;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.exceptions.CommitStateUnknownException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Queues;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.relocated.com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.iceberg.util.Tasks;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Functionality used by RewriteDataFile Actions from different platforms to handle commits.
+ */
+public class RewriteDataFilesCommitUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(RewriteDataFilesCommitUtil.class);
+
+  private final Table table;
+
+  public RewriteDataFilesCommitUtil(Table table) {
+    this.table = table;
+  }
+
+  /**
+   * Perform a commit operation on the table adding and removing files as
+   * required for this set of file groups
+   * @param fileGroups fileSets to commit
+   */
+  public void commitFileGroups(Set<RewriteFileGroup> fileGroups) {
+    Set<DataFile> rewrittenDataFiles = fileGroups.stream()
+        .flatMap(fileGroup -> fileGroup.rewrittenFiles().stream()).collect(Collectors.toSet());
+
+    Set<DataFile> newDataFiles = fileGroups.stream()
+        .flatMap(fileGroup -> fileGroup.addedFiles().stream()).collect(Collectors.toSet());
+
+    table.newRewrite()
+        .rewriteFiles(rewrittenDataFiles, newDataFiles)
+        .commit();
+  }
+
+  /**
+   * Clean up a specified file set by removing any files created for that operation, should
+   * not throw any exceptions
+   * @param fileGroup fileSet to clean
+   */
+  public void abortFileGroup(RewriteFileGroup fileGroup) {
+    if (fileGroup.addedFiles() == null) {
```

Comment:
I think this is never null in `RewriteFileGroup` (which is correct) so we may drop this condition?

Reply:
Changed this to a check state since as @rdblue noted this should never be called on a file group which hasn't already successfully gone through rewriting.